engine/runtime/src/main/java/org/enso/interpreter/runtime/state/HasContextEnabledNode.java

```diff
@@ -0,0 +1,50 @@
+package org.enso.interpreter.runtime.state;
+
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.GenerateUncached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.atom.AtomConstructor;
+
+/**
+ * A node representing the functionality done by {@code Standard.Base.Runtime.Context.is_enabled}.
+ */
+@GenerateUncached
+public abstract class HasContextEnabledNode extends Node {
+
+  public static HasContextEnabledNode getUncached() {
+    return HasContextEnabledNodeGen.getUncached();
+  }
+
+  public static HasContextEnabledNode create() {
+    return HasContextEnabledNodeGen.create();
+  }
+
+  /**
+   * Returns true if the context represented by the given {@code runtimeCtxCtor} is enabled in the
+   * given {@code executionEnvironment}.
+   *
+   * @param runtimeCtxCtor Constructor of {@code Standard.Base.Runtime.Context}.
+   */
+  public abstract boolean executeHasContextEnabled(
+      ExecutionEnvironment executionEnvironment, AtomConstructor runtimeCtxCtor);
+
+  @Specialization(guards = "executionEnvironment == cachedEnv", limit = "3")
+  boolean doIt(
+      ExecutionEnvironment executionEnvironment,
+      AtomConstructor runtimeCtxCtor,
+      @Cached("executionEnvironment") ExecutionEnvironment cachedEnv) {
+    var ensoCtx = EnsoContext.get(this);
+    var contextBuiltin = ensoCtx.getBuiltins().context();
+    if (runtimeCtxCtor == contextBuiltin.getInput()) {
+      return cachedEnv.permissions.input();
+    } else if (runtimeCtxCtor == contextBuiltin.getOutput()) {
+      return cachedEnv.permissions.output();
+    } else if (runtimeCtxCtor == contextBuiltin.getDataflowStackTrace()) {
+      return cachedEnv.permissions.dataflowStacktrace();
+    } else {
+      throw ensoCtx.raiseAssertionPanic(this, "Unknown context: " + runtimeCtxCtor, null);
```

Comment:
String concatenation cannot be on fast path: https://github.com/enso-org/enso/actions/runs/11387183014/job/31745369546?pr=11153#step:7:1631

Reply:
Thanks for spotting this. Added `trasnferToInterpreter` in d3f4c8c812728cf8b5a60a91f664f9aba32a93e3